### PR TITLE
[KBV-109] Add cucumber tests 

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

Add cucumber tests to automatically test the frontend app, using mocked data. Different paths through the application can be selected by using the `scenario-id` header.

### Why did it change

Using browser level tests provides a way to test the rendered page as displayed to the user. This allows testing of expected paths through the system, and validation messages.

### Issue tracking

- [KBV-109](https://govukverify.atlassian.net/browse/KBV-109)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: test/application/pages/question.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  async answerIncorrectly(value) {
+    await this.page
+      .locator(`input[type="radio"]:not([value="${value}"])`)
+      .first()
+      .check();
+  }
+
+  // async getStarted() {
```
Comment (MattBidewell) [inline test/application/pages/question.js:56]: Should we/do we remove commented code if its not being used?

**Review by MattBidewell — COMMENTED** [review]
LGTM

File: test/application/pages/question.js (hunk 1 above)
Comment (merlinc) [reply test/application/pages/question.js:56]: Yes! :)

File: test/application/support/setup.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const { Before, BeforeAll, AfterAll, After } = require("@cucumber/cucumber");
+const { chromium } = require("playwright");
+const axios = require("axios");
+
+const users = {
```
Comment (MattBidewell) [inline test/application/support/setup.js:5]: These users dont match the users in test/application/support/env.js. Would this be a problem/change needed going forward?

Comment (merlinc) [reply test/application/support/setup.js]: Good catch - I've made the two files consistent,  but as you can see they're not being used properly yet.

At the moment they're used to mentally separate out the participants in the test - I'm trying to take a few pointers from https://cucumber.io/docs/bdd/better-gherkin/

Once these can integrate with the backend and/or the stub, then I'm intending on tweaking the code to use the values given.

File: test/browser/step_definitions/kbv.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+});
+
+Then(/^they should see validation messages$/, async function () {
+  const questionPage = new QuestionPage(this.page);
+
+  const errorSummary = await questionPage.getErrorSummary();
+
+  expect(errorSummary).to.include(questionPage.getNotAnsweredErrorMessage());
+
+  // assert.equal(errorSummary, questionPage.getNotAnsweredErrorMessage());
```
Comment (MattBidewell) [inline test/browser/step_definitions/kbv.js:70]: Lets remember to deleted unused code 😁

**Review by MattBidewell — APPROVED** [review]
LGTM

**Review by MattBidewell — APPROVED** [review]

**Pull request merged**